src/overlays/actors/ovl_Bg_Ydan_Hasi/z_bg_ydan_hasi.c

```diff
@@ -21,23 +27,180 @@ const ActorInit Bg_Ydan_Hasi_InitVars = {
+    return waterBox + 0x1;
+}
+
+void BgYdanHasi_DecWaterTimer(BgYdanHasi* this, GlobalContext* globalCtx) {
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    func_8002F994(&this->dyna.actor, this->timer);
+    if (this->timer == 0) {
+        this->actionFunc = BgYdanHasi_MoveWater;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/func_808BE690.s")
+void BgYdanHasi_SetupThreeBlocks(BgYdanHasi* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, this->unk_168) != 0) {
+        this->timer = 260;
+        this->dyna.actor.draw = BgYdanHasi_Draw;
+        this->actionFunc = BgYdanHasi_UpdateThreeBlocks;
+        func_800800F8(globalCtx, 0xBE0, 0x1E, &this->dyna.actor, 0);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/func_808BE7C8.s")
+void BgYdanHasi_UpdateThreeBlocks(BgYdanHasi* this, GlobalContext* globalCtx) {
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/func_808BE810.s")
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    if (this->timer == 0) {
+        if (Math_ApproxF(&this->dyna.actor.posRot.pos.y, this->dyna.actor.initPosRot.pos.y, 3.0f) != 0) {
+            Flags_UnsetSwitch(globalCtx, this->unk_168);
+            this->dyna.actor.draw = NULL;
+            this->actionFunc = BgYdanHasi_SetupThreeBlocks;
+            return;
+        }
+        func_8002F948(&this->dyna.actor, NA_SE_EV_ELEVATOR_MOVE - SFX_FLAG);
+        return;
+    }
+    if (Math_ApproxF(&this->dyna.actor.posRot.pos.y, this->dyna.actor.initPosRot.pos.y + 120.0f, 3.0f) == 0) {
+        func_8002F948(&this->dyna.actor, NA_SE_EV_ELEVATOR_MOVE - SFX_FLAG);
+        return;
+    }
+    func_8002F994(&this->dyna.actor, this->timer);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/func_808BE8DC.s")
+void BgYdanHasi_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgYdanHasi* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/func_808BE930.s")
+    this->actionFunc(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/func_808BE99C.s")
+void BgYdanHasi_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    GraphicsContext* gfxCtx;
+    Gfx* dispRefs[4];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/BgYdanHasi_Update.s")
+    if (thisx->params == 0 || thisx->params == 2) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Ydan_Hasi/BgYdanHasi_Draw.s")
+        Gfx_DrawDListOpa(globalCtx, D_808BEC24[thisx->params]);
+        return;
+    }
+    gfxCtx = globalCtx->state.gfxCtx;
+    Graph_OpenDisps(dispRefs, globalCtx->state.gfxCtx, "../z_bg_ydan_hasi.c", 577);
+    func_80093D84(globalCtx->state.gfxCtx);
+    gSPSegment(gfxCtx->polyXlu.p++, 0x08,
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, (0 - globalCtx->gameplayFrames & 0x7F),
```

Comment:
I'm pretty sure we want % 128 instead of & 0x7F for these, but nice job matching a Gfx_TwoTexScroll which I seem to suck at :P

Reply:
It wont match when doing % 128.  That Gfx_TwoTexScroll was completely done by the glank tool. And Figs help.

Reply:
gfxdis doesnt do anything with two tex scroll, that was mips2c. it should work with mod but if not thats interesting

Reply:
![Capture](https://user-images.githubusercontent.com/35883445/87734516-d3c7c180-c7a0-11ea-9bb6-3a4b404efc40.PNG)
Here it is when doing `globalCtx->gameplayFrames % 128`

Reply:
Maybe I'll try & 0x7F on my issue then... :P